spec/lib/supplier_import_row_spec.rb

```diff
@@ -0,0 +1,95 @@
+require 'rails_helper'
+require './lib/supplier_import_row'
+
+RSpec.describe SupplierImportRow do
+  describe '#initialize' do
+    it 'throws an error if the hash is missing required fields' do
+      data = {
+        frameworkreference: 'RM3756'
+      }
+
+      expect { SupplierImportRow.new(data) }.to raise_error('SupplierImportRow::MissingKey')
+    end
+  end
+
+  describe '#import!' do
+    it 'creates a new supplier and adds it to the framework' do
+      framework = FactoryBot.create(:framework, short_name: 'RM3756')
+
+      data = {
```

Comment:
given that this same data structure is used three times, it's probably worth pulling it out to a method or `let`